traffic_portal/app/src/modules/private/certExpirations/list/index.js

```diff
@@ -32,19 +32,6 @@ module.exports = angular.module('trafficPortal.private.certExpirations.list', []
 							},
 							tableName: function() {
 								return 'cert-expirations';
-							},
-							filter: function() {
-								return null;
-							},
-							deliveryservices: function(deliveryServiceService) {
-								return deliveryServiceService.getDeliveryServices();
-							},
```

Comment:
this is most of what `resolve` values are for: you get a promise that asynchronously outputs data, and Angular handles resolving it to an actual value injected into your controller at component initialization time. If you keep `deliveryservices` here, you can input it as an injected dependency like it used to be, and instead of a function like
```javascript
function init() {
    doSomething();
}
init();
```
(which, incidentally is in no way any different than just calling `doSomething` immediately - the only real differences are `this` binding and the values of some global variables that no one should ever use like `name` and `arguments`)

you can just directly construct the map immediately like:
```javascript
	$scope.dsXmlToIdMap = new Map(deliveryservices.map(ds=>[ds.xmlId, ds.id]));
```
no extra function wrapping or deferred calling or `.then`-ing a promise necessary at all.